common/src/main/java/com/wynntils/screens/base/WynntilsGridLayoutScreen.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.base;
+
+import com.wynntils.core.consumers.screens.WynntilsScreen;
+import com.wynntils.screens.base.widgets.TextInputBoxWidget;
+import net.minecraft.network.chat.Component;
+import org.lwjgl.glfw.GLFW;
+
+public abstract class WynntilsGridLayoutScreen extends WynntilsScreen implements TextboxScreen {
+    private static final float GRID_DIVISIONS = 64f;
+    // Certain elements require static 20 height or button textures will break
+    protected static final int BUTTON_HEIGHT = 20;
+    protected float dividedHeight;
+    protected float dividedWidth;
+    private TextInputBoxWidget focusedTextInput;
+    /*
+    Some notes on element alignment and sizing:
+    All elements should be positioned dynamically relative to window size. That is; you must use this.height and
+    this.width, plus any multiplier and optionally, +/- 1 to make small tweaks. For example, these are valid:
+    (int) (dividedWidth * 36)
+    (int) (dividedWidth * 57) - 1
+    (this.width / 2)
+    (end of another element) + x
+    And these are invalid:
+    100
+    (int) (dividedWidth * 36) + 4
+    (int) (this.width / 2 + 199)
+    The reason for all of this is gui scale. Height and width will be different depending on the gui scale, and
+    the above method will ensure that the gui is always aligned somewhat correctly.
+
+    Also note that button height (and related elements), must be 20 or the texture will break.
+    Element sizing isn't nearly as picky as alignment, but it'll still be better to keep static modifiers (eg. +2) to a
+    minimum.
+     */
+
+    protected WynntilsGridLayoutScreen(Component component) {
+        super(component);
+    }
+
+    @Override
+    protected void doInit() {
+        dividedHeight = this.height / GRID_DIVISIONS;
+        dividedWidth = this.width / GRID_DIVISIONS;
+    }
+
+    @Override
+    public boolean charTyped(char codePoint, int modifiers) {
+        return (focusedTextInput != null && focusedTextInput.charTyped(codePoint, modifiers))
```

Comment:
Shouldn't it be `WynntilsScreen` handle implementing `TextboxScreen` + the input methods that need to be overwritten?

Reply:
I don't think so, there are a lot of WynntilsScreens that don't have anything to do with textboxes

Reply:
It's still weird to have the code handling this in this class.. Even if a lot of screens don't use it in WynntilScreen, it still could be there though? Or maybe a new superclass, `WynntilsTextboxScreen`? I see no point in making a new class though, if WynntilsScreen can handle it. Of  course a few of the overriden methods would need to be adjusted to call super properly.

Reply:
If we could solve this comment, then we can probably merge this, I don't see any issues otherwise.